### PR TITLE
FLAG-67: Fix rendering of flags via simple objects that is currently broken in platform version greater than 2.5.* 

### PR DESCRIPTION
**Pull request by ManojLL**
## Description of what I changed

I added a key for the object return from PatientflagsDashboardFragmentController to the UI and fixed the flags rendering issue on the UI

image

## Issue I worked on

see https://openmrs.atlassian.net/browse/FLAG-67

## Checklist: I completed these to help reviewers :)

- [x] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [ ] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [x] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [x] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [x] My pull request is **based on the latest changes** of the master branch.
  
  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

**Comment by wikumChamith** [comment]
@ManojLL this is not working form me.

Screenshot from 2024-06-30 10-12-48
Screenshot from 2024-06-30 10-12-43

**Comment by ManojLL** [comment]
@wikumChamith  Flags are showing based on roles, could you please check your flags tags contain roles ?

**Comment by wikumChamith** [comment]
> @wikumChamith Flags are showing based on roles, could you please check your flags tags contain roles ?

Shouldn't both the legacy-ui and O2 behave in the same way?

**Comment by ManojLL** [comment]
> > @wikumChamith Flags are showing based on roles, could you please check your flags tags contain roles ?
> 
> Shouldn't both the legacy-ui and O2 behave in the same way?

no, Legacy UI shows all triaged flags to any user.

**Comment by wikumChamith** [comment]
> @wikumChamith Flags are showing based on roles, could you please check your flags tags contain roles ?

@dkayiwa what is your opinion on this, Should we display all the flags here or show the flags based on roles?

**Comment by dkayiwa** [comment]
How about just maintaining how it has always been?

**Comment by wikumChamith** [comment]
@ManojLL I am not seeing the styles related to the priority on the O2 UI.

Screenshot from 2024-07-10 16-02-13
Screenshot from 2024-07-10 16-02-04

**Comment by ManojLL** [comment]
@wikumChamith fix the style issue

File: omod/src/main/webapp/fragments/patientflagsDashboard.gsp [hunk 1]
```diff
@@ -15,7 +15,7 @@
                 if (!response) {
                     ${ ui.message("coreapps.none ") }
                 } else {
-                    jq("#flags").html(response.replace("{=", "").replace("}", ""));
```
Comment (dkayiwa) [inline omod/src/main/webapp/fragments/patientflagsDashboard.gsp:18]: @wikumChamith does this work on lower platform versions?

Comment (wikumChamith) [reply omod/src/main/webapp/fragments/patientflagsDashboard.gsp:18]: I tested this on Reference Application 2.11.0 but this is not working there.

Comment (dkayiwa) [reply omod/src/main/webapp/fragments/patientflagsDashboard.gsp:18]: @wikumChamith meaning that this change breaks the module for those who are running lower platform versions.

**Pull request merged**